langchain4j-redis/src/main/java/dev/langchain4j/store/embedding/redis/RedisEmbeddingStore.java

```diff
@@ -295,8 +310,16 @@ public Builder metadataKeys(Collection<String> metadataKeys) {
             return this;
         }
 
+        /**
+         * @param schemaFieldMap Metadata schemaField, the key set must be the same as {@link Builder#metadataKeys(Collection)} (optional)
+         */
+        public Builder schemaFiledMap(Map<String, SchemaField> schemaFieldMap) {
```

Comment:
```suggestion
        public Builder metadataKeys(Map<String, SchemaField> metadataKeys) {
```
It would probably feel a bit strange for the user to provide the same information (metadata keys) twice. And there will be a chance that `metadataKeys` and `schemaFiledMap` are not in sync.

What about having a `metadataKeys` method instead where user can specify not only the keys but also types?

So this:
```java
RedisEmbeddingStore.builder()
                ...
                .metadataKeys(Set.of("one", "two"))
                .schemaFiledMap(Map.of("one", NumericField.of(...), "two", NumericField.of(...)))
                .build();
```
Will be simplified to this:
```java
RedisEmbeddingStore.builder()
                ...
                .schemaFiledMap(Map.of("one", NumericField.of(...), "two", NumericField.of(...)))
                .build();
```

Reply:
@langchain4j Done. I make `metadatakeys` deprecated and will use `TextField` as default SchemaField if users still use this method.

Reply:
I am not sure about `schemaFiledMap` method name, perharps we should keep `metadataKeys` or something like `metadataFields`? It sohuld have "metadata" in the name for sure